gcloud-java-dns/src/main/java/com/google/gcloud/dns/spi/RpcBatch.java

```diff
@@ -0,0 +1,117 @@
+
+import java.util.Map;
+
+/**
+ * An interface for the collection of batch operations.
+ */
+public interface RpcBatch {
+
+  /**
+   * An interface for batch callbacks.
+   */
+  interface Callback<T> {
+
+    /**
+     * This method will be called upon success of the batch operation.
+     */
+    void onSuccess(T response);
+
+    /**
+     * This method will be called upon failure of the batch operation.
+     */
+    void onFailure(GoogleJsonError googleJsonError);
+  }
+
+  /**
+   * Adds a call to "list zones" to the batch with the provided {@code callback} and {@code
+   * options}.
+   */
+  void addListZones(Callback<ManagedZonesListResponse> callback, Map<DnsRpc.Option, ?> options);
+
+  /**
+   * Adds a call to "create zone" to the batch with the provided {@code callback} and {@code
+   * options}.
+   */
+  void addCreateZone(ManagedZone zone, Callback<ManagedZone> callback,
+      Map<DnsRpc.Option, ?> options);
+
+  /**
+   * Adds a call to "get zone" to the batch with the provided {@code callback} and {@code options}.
+   * The zone to be retrieved is identified by {@code zoneName}.
+   */
+  void addGetZone(String zoneName, Callback<ManagedZone> callback, Map<DnsRpc.Option, ?> options);
+
+  /**
+   * Adds a call to "get project" to the batch with the provided {@code callback} and {@code
+   * options}.
+   */
+  void addGetProject(Callback<Project> callback, Map<DnsRpc.Option, ?> options);
+
+  /**
+   * Adds a call to "delete zone" to the batch with the provided {@code callback} and {@code
+   * options}.
+   */
+  void addDeleteZone(String zoneName, Callback<Void> callback);
+
+  /**
+   * Adds a call to "list record sets" to the batch with the provided {@code callback} and {@code
+   * options}. The zone whose record sets are to be listed is identified by {@code zoneName}.
+   */
+  void addListRecordSets(String zoneName, Callback<ResourceRecordSetsListResponse> callback,
+      Map<DnsRpc.Option, ?> options);
+
+  /**
+   * Adds a call to "list change requests" to the batch with the provided {@code callback} and
+   * {@code options}. The zone whose change requests are to be listed is identified by {@code
+   * zoneName}.
+   */
+  void addListChangeRequests(String zoneName, Callback<ChangesListResponse> callback,
+      Map<DnsRpc.Option, ?> options);
+
+  /**
+   * Adds a call to "get change request" to the batch with the provided {@code callback} and {@code
+   * options}. The change request to be retrieved is identified by {@code changeRequestId} and the
+   * zone to which the change request was applied is identified by {@code zoneName}.
+   */
+  void addGetChangeRequest(String zoneName, String changeRequestId, Callback<Change> callback,
+      Map<DnsRpc.Option, ?> options);
+
+  /**
+   * Adds a call to "apply change request" to the batch with the provided {@code callback} and
+   * {@code options}. The zone to which the change request should be applied is identified by {@code
+   * zoneName}.
```

Comment:
Nit: slight break in pattern established above by not mentioning `change` input.